### DISCUSSION TITLE
How to iterate over a Project Class to access the HW confic data in the Step7 Project?

### DISCUSSION THREAD

**Discussion by OleKristianLieNOV** [dotnetprojects/DotNetSiemensPLCToolBoxLibrary#201] [Q&A]
Hello. 
I am currently learning C# so if the answer is obvious please forgive me. 

**My issue:**
When executing Project.AllFolders in realtime, the content of the Project.AllFolders is not being loaded and my application continues without data and crashes. When running in debug mode and manually pausing for that step then it loads Project.AllFolders into memory and executes just fine.

**My code:**
```
Project project = Projects.LoadProject(filePath, showDeleted: false, null);
var allFolders =  project.AllFolders;

foreach (var folder in allFolders)
{
    string folderName = folder.Name;
    if (folderName != null && folderName.StartsWith("CPU "))
    {
        CPUFolder cpu = folder as CPUFolder;
        string cpuType = cpu.CpuType.ToString();
        WriteLine("The CPU type is " + cpuType);
        WriteLine();
    }
}
```

**Things I have tested that did not work:**
Iterating over the Project instead of Project.AllFolders, but that does not work.
> 'Project' does not contain a public instance or extension definition for 'GetEnumerator'

Adding a async method that reads Project.AllFolders and awaiting that, but that sent me back to the original issue where Project.AllFoldes is not loaded when running normally, but manually waiting in debug mode it is loaded.

Thank you for taking the time to review and please let me know if any more information is needed or you know of a solution!

**Comment by jogibear9988** [comment]
Sorry didn't see this issue until now.
I don't know what your issue is. The code is sync so it should be the same in debug as without. Could you create a small sample app?
